src/testsuite/vec_perf_i128.c

```diff
@@ -124,4 +157,81 @@ int timed_mulluq (void)
   return rc;
 }
 
+#pragma GCC optimize ("no-unroll-loops")
+//#define __DEBUG_PRINT__
+int
+timed_muludq (void)
+{
+  vui128_t i, j, k;
+  int ii, iii;
+  int rc = 0;
+
+  i = (vui128_t) c_ten;
+  iii = 2;
+#ifdef __DEBUG_PRINT__
+  print_vint128 ("init i=10", (vui128_t) i);
+#endif
+
+  for (ii = 1; ii < 7; ii++)
+    {
+      k = vec_muludq (&j, (vui128_t) i, (vui128_t) i);
+
+#ifdef __DEBUG_PRINT__
+      if (ii < 6)
+	{
+	  printf ("iii=%d ", iii);
+	  print_vint128 ("10Eiii=", k);
+	  rc += check_vuint128 ("vec_muludq:", k, vtipowof10[iii]);
+	}
+#endif
+      i = k;
+      iii = iii + iii;
+    }
+  rc += check_vuint128x ("vec_muludqx k:", k, (vui128_t) ten_64l);
+  rc += check_vuint128x ("vec_muludqx j:", j, (vui128_t) ten_64h);
+
+  return rc;
+}
+
+//#define __DEBUG_PRINT__
+/* Older compilers ignore the "no-unroll-loops" pragma and make it
+   hard to isolate each iteration of vec_muludq.  This version
+   calls __test_muludq from vec_int128_dummy.c as a work around.  */
+extern vui128_t
+__test_muludq (vui128_t *mh, vui128_t a, vui128_t b);
+
+int
+timed_muludqx (void)
+{
+  vui128_t i, j, k;
+  int ii, iii;
+  int rc = 0;
+
+  i = (vui128_t) c_ten;
+  iii = 2;
+#ifdef __DEBUG_PRINT__
+  print_vint128 ("init i=10", (vui128_t) i);
+#endif
+
+  for (ii = 1; ii < 7; ii++)
+    {
+      k = __test_muludq (&j, (vui128_t) i, (vui128_t) i);
+
+#ifdef __DEBUG_PRINT__
+      if (ii < 6)
+	{
+	  printf ("iii=%d ", iii);
+	  print_vint128 ("10Eiii=", k);
+	  rc += check_vuint128 ("vec_muludq:", k, vtipowof10[iii]);
+	}
+#endif
+      i = k;
+      iii = iii + iii;
+    }
+  rc += check_vuint128x ("vec_muludqx k:", k, (vui128_t) ten_64l);
+  rc += check_vuint128x ("vec_muludqx j:", j, (vui128_t) ten_64h);
+
+  return rc;
+}
+
```

Comment:
total nit: extra blank line here.
now you know I read all the way to the end.  ;-)